### PR TITLE
Improve clients layout

### PR DESCRIPTION
**Pull request by thibaultamartin**
Part of #1538 

Left to do:
- [x] Add a title "All known clients" and figure out what's the best place for it (aligned with "Clients" title? aligned with table?)
- [x] Add an introductory text in the "All known clients" section. e.g. "The Matrix ecosystem is vibrant: whatever platform you use, there are good chances someone has developed a client for it. Here we list all the clients the Foundation is aware of, with a maturity score and a list of important features they support or not to help you pick the best for you."
- [x] Make the store links less cramped/better delimited
- [x] Centring the ticks and crosses

### CODE REVIEW

**Comment by thibaultamartin** [comment]
Not sure if we should vertically align the text in the cells? Also not really sure what's the best way to give some breathing space to the platform links. Maybe pill-buttons like in the featured clients?

**Comment by jplatte** [comment]
My 2c: Yes to pills, no to vertical centering.

**Comment by thibaultamartin** [comment]
Screenshot 2022-11-03 at 08-49-18 🛠️ Matrix org - Clients

Pills seem to take quite a bit of vertical real estate here, even with not a lot of padding :/

**Comment by jplatte** [comment]
We should set a larger width for the platforms column, otherwise looks good IMO.

File: templates/clients.html [hunk 1]
```diff
@@ -1,171 +1,157 @@
...
-        </table>
+
+    <h2>All known clients</h2>
+    <p>
+        The Matrix ecosystem is vibrant: whatever platform you use, there are
+        good chances someone has developed a client for it. Here we list all the
+        clients the Foundation is aware of, with a maturity score and a list of
+        important features they support or not to help you pick the best for
+        you.
+    </p>
```
Comment (thibaultamartin) [inline templates/clients.html:46]: I'm not too happy about this being in the template, but I'm not really sure what's the best place to put it either

**Comment by thibaultamartin** [comment]
Screenshot 2022-11-03 at 14-38-09 🛠️ Matrix org - Clients

It seems acceptable as is

File: sass/_clients.scss [hunk 2]
```diff
@@ -1,26 +1,123 @@
...
+    overflow-x: auto;
+}
+
+.featured-client-card {
+    display: flex;
+    flex-direction: column;
+    flex: 0 0 250px;
+    border-style: solid;
+    border-width: 1px;
+    border-color: rgba(43, 140, 201, .4);
```
Comment (jplatte) [inline sass/_clients.scss:13]:
```suggestion
    border: 1px solid rgba(43, 140, 201, .4);
```

File: sass/_clients.scss [hunk 3]
```diff
@@ -1,26 +1,123 @@
...
-        overflow-x: auto;
-        width: unset;
+        margin: 0 auto;
+    }
+
+    h3 {
+        text-align: center;
+    }
+
+    .stores-links {
```
Comment (jplatte) [inline sass/_clients.scss:28]: Should "stores" be renamed "platforms" here for consistency?

File: sass/_clients.scss [hunk 4]
```diff
@@ -1,26 +1,123 @@
...
+.platform-pill {
+    margin: .2em;
+    border-radius: 2em;
+    padding: 0 .6em;
+    color: #fff;
+    background-color: rgba(43, 140, 201, .8);
+}
+
+.platform-pill:hover {
+    background-color: rgba(43, 140, 201, 1);
```
Comment (jplatte) [inline sass/_clients.scss:45]: This can use `#rrggbb` notation, right?

File: templates/clients.html [hunk 5]
```diff
@@ -1,171 +1,119 @@
...
-                </div>
-            </div>
+<div class="content">
+    <header>
+        <h1>{{ page.title }}</h1>
+    </header>
+    <p>{{ page.content | safe }}</p>
+</div>
+<div class="all-clients">
+    <!-- Table -->
```
Comment (jplatte) [inline templates/clients.html:11]:
```suggestion
```

File: templates/shortcodes/clients_showcase.html [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+            {% endif %}
+            {% if client.packages.f_droid.app_id %}
+            <a href="https://f-droid.org/packages/{{ client.packages.f_droid.app_id }}" class="platform-pill">F-Droid</a>
+            {% endif %}
+            {% if client.packages.flathub.app_id %}
+            <a href="https://flathub.org/apps/details/{{ client.packages.flathub.app_id }}" class="platform-pill">Flathub</a>
+            {% endif %}
+            {% if client.packages.other_url %}
+            <a href="{{ client.packages.other_url }}" class="platform-pill">Web</a>
+            {% endif %}
```
Comment (jplatte) [inline templates/shortcodes/clients_showcase.html:24]: This is duplicated between the card and table code, right? Maybe we should move it out into a macro.

File: templates/clients.html [hunk 7]
```diff
@@ -1,171 +1,119 @@
...
+            {% else %}
+            <span class="cross">✗</span>
+            {% endif %}
+        </span>
+        <span class="feature">
+            {% if client.features.voip_1to1 %}
+            <span class="checkmark">✓</span>
+            {% else %}
+            <span class="cross">✗</span>
+            {% endif %}
```
Comment (jplatte) [inline templates/clients.html:72]: This boolean-to-checkmark-or-cross thing also seems macro-worthy to me.

**Review by jplatte — COMMENTED** [review]
Feel free to pass off some of the template work to me if you don't want to do it.

**Review by jplatte — APPROVED** [review]

**Pull request merged**